src/silly_sprites/events.hpp

```diff
@@ -0,0 +1,31 @@
+#include <string>
+#include <type_traits>
+#include <variant>
+#include <functional>
+
+namespace sly::event {
+    // add new events here
+    struct CollisionEvent { };
+    struct SoundEvent { };
+    struct MessageEvent {
+    private:
+        std::string m_message;
+
+    public:
+        MessageEvent(std::string message) : m_message{ message } { }
+
+        [[nodiscard]] std::string message() const {
+            return m_message;
+        }
+    };
+
+    template<typename T, typename... Ts>
+    concept IsAnyOf = std::disjunction_v<std::is_same<T, Ts>...>;
+    template<typename... T>
+    using VariantCallbacks = std::variant<std::function<void(T const&)>...>;
+
+    template<typename T>
+    concept Event = IsAnyOf<T, CollisionEvent, SoundEvent, MessageEvent>;              // add new events here
+    using EventCallbacks = VariantCallbacks<CollisionEvent, SoundEvent, MessageEvent>; // add new events here
```

Comment:
Some sick magic going on here, I like it!
![giphy](https://github.com/mgerhold/silly-sprites/assets/51532668/678a0a54-da36-4d52-b0b7-37babe98e567)

Reply:
chat did things. LUL